### PR TITLE
Protocol Buffer update for release 1.15.x at Wed Feb 24 09:15:41 UTC 2021

### PR DESCRIPTION
**Pull request by Corona-Warn-App-Technical-User**

### CODE REVIEW

**Comment by harambasicluka** [comment]
We agreed that this can be merged in 1.15 as we don't care about the protobuf files even if we only ship it with 1.16.

**Comment by d4rken** [comment]
```proto
Execution failed for task ':Server-Protocol-Buffer:generateProto'.
> protoc: stdout: . stderr: internal/v2/event_registration_parameters.proto:17:9: "de.rki.coronawarnapp.server.protocols.internal.v2.TransmissionRiskValueMapping.transmissionRiskLevel" is already defined in file "internal/v2/risk_calculation_parameters.proto".
  internal/v2/event_registration_parameters.proto:18:10: "de.rki.coronawarnapp.server.protocols.internal.v2.TransmissionRiskValueMapping.transmissionRiskValue" is already defined in file "internal/v2/risk_calculation_parameters.proto".
  internal/v2/event_registration_parameters.proto:16:9: "de.rki.coronawarnapp.server.protocols.internal.v2.TransmissionRiskValueMapping" is already defined in file "internal/v2/risk_calculation_parameters.proto".
  internal/v2/app_config_android.proto: Import "internal/v2/event_registration_parameters.proto" was not found or had errors.
  internal/v2/app_config_android.proto:37:3: "EventRegistrationParameters" is not defined.
```

**Comment by mlenkeit** [comment]
@d4rken the error that you posted should be gone. Are you sure the build is still failing because of the changes of this PR?

**Comment by d4rken** [comment]
```java
e: /home/circleci/project/Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/analytics/Analytics.kt: (15, 60): Unresolved reference: PpaDataRequestAndroid
e: /home/circleci/project/Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/analytics/Analytics.kt: (57, 32): Unresolved reference: PpaDataRequestAndroid
e: /home/circleci/project/Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/analytics/server/DataDonationAnalyticsApiV1.kt: (4, 60): Unresolved reference: PpaDataRequestAndroid
e: /home/circleci/project/Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/analytics/server/DataDonationAnalyticsApiV1.kt: (17, 28): Unresolved reference: PpaDataRequestAndroid
e: /home/circleci/project/Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/analytics/server/DataDonationAnalyticsServer.kt: (5, 60): Unresolved reference: PpaDataRequestAndroid
e: /home/circleci/project/Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/analytics/server/DataDonationAnalyticsServer.kt: (15, 60): Unresolved reference: PpaDataRequestAndroid
e: /home/circleci/project/Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/safetynet/AttestationContainer.kt: (5, 60): Unresolved reference: PpacAndroid
e: /home/circleci/project/Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/safetynet/AttestationContainer.kt: (12, 41): Unresolved reference: PpacAndroid
e: /home/circleci/project/Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/safetynet/AttestationContainer.kt: (13, 17): Unresolved reference: PpacAndroid
e: /home/circleci/project/Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/safetynet/AttestationContainer.kt: (14, 13): Unresolved reference: salt
e: /home/circleci/project/Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/safetynet/AttestationContainer.kt: (15, 13): Unresolved reference: safetyNetJws
e: /home/circleci/project/Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/safetynet/DeviceAttestation.kt: (4, 60): Unresolved reference: PpacAndroid
e: /home/circleci/project/Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/safetynet/DeviceAttestation.kt: (27, 36): Unresolved reference: PpacAndroid
e: /home/circleci/project/Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/survey/server/SurveyServer.kt: (30, 14): Overload resolution ambiguity: 
public open fun setAuthentication(p0: PPACAndroidProto.PPACAndroid!): de.rki.coronawarnapp.server.protocols.internal.ppdd.EdusOtpRequestAndroid.EDUSOneTimePasswordRequestAndroid.Builder! defined in de.rki.coronawarnapp.server.protocols.internal.ppdd.EdusOtpRequestAndroid.EDUSOneTimePasswordRequestAndroid.Builder
public open fun setAuthentication(p0: de.rki.coronawarnapp.server.protocols.internal.ppdd.PPACAndroidProto.PPACAndroid.Builder!): de.rki.coronawarnapp.server.protocols.internal.ppdd.EdusOtpRequestAndroid.EDUSOneTimePasswordRequestAndroid.Builder! defined in de.rki.coronawarnapp.server.protocols.internal.ppdd.EdusOtpRequestAndroid.EDUSOneTimePasswordRequestAndroid.Builder

```

File: Server-Protocol-Buffer/src/main/proto/internal/ppdd/ppac_android.proto [hunk 1]
```diff
@@ -3,9 +3,11 @@ syntax = "proto3";
 package de.rki.coronawarnapp.server.protocols.internal.ppdd;
 
 
+option java_outer_classname = "PPACAndroidProto";
```
Comment (d4rken) [inline Server-Protocol-Buffer/src/main/proto/internal/ppdd/ppac_android.proto:6]: Well the issue is that this renames the generated protobuf files.

File: Server-Protocol-Buffer/src/main/proto/internal/ppdd/ppa_data_request_android.proto [hunk 2]
```diff
@@ -5,9 +5,11 @@ package de.rki.coronawarnapp.server.protocols.internal.ppdd;
 import "internal/ppdd/ppac_android.proto";
 import "internal/ppdd/ppa_data.proto";
 
+option java_outer_classname = "PPADataRequestAndroidProto";
```
Comment (d4rken) [inline Server-Protocol-Buffer/src/main/proto/internal/ppdd/ppa_data_request_android.proto:8]: Same here, renames the generated files.

File: Server-Protocol-Buffer/src/main/proto/internal/ppdd/ppac_android.proto (hunk 1 above)
Comment (mlenkeit) [reply Server-Protocol-Buffer/src/main/proto/internal/ppdd/ppac_android.proto:6]: I was hoping this is actually only relevant for Java... i'll fix it

**Comment by mlenkeit** [comment]
image
🎉

**Review by BMItr — APPROVED** [review]
👌

**Review by d4rken — APPROVED** [review]

**Pull request merged**